capytaine/io/xarray.py

```diff
@@ -238,15 +239,44 @@ def assemble_dataset(results: Sequence[LinearPotentialFlowResult],
     """
     dataset = xr.Dataset()
 
+    if hasattr(results, '__iter__'):
+        try:
+            if 'capytaine' in results[0].__module__:
+                bemio_import = False
+            else:
+                raise TypeError('results must be either of type LinearPotentialFlowResult or a bemio.io object')
+        except:
+            raise TypeError('results must be either of type LinearPotentialFlowResult or a bemio.io object')
+
+    else:
+        try:
+            if 'bemio.io' in results.__module__:
+                bemio_import = True
+            else:
+                raise TypeError('results must be either of type LinearPotentialFlowResult or a bemio.io object')
+        except:
+            raise TypeError('results must be either of type LinearPotentialFlowResult or a bemio.io object')
```

Comment:
Suggestion: avoid the repetition or the error message in the code by storing it in a variable:
```
error = TypeError("blabla")
if ...:
    raise error
else:
    raise error
```

Reply:
Agreed, this makes it much more readable. Fixed in commit 0f16541.